tests/lib/rules/prefer-stateless-function.js

```diff
@@ -345,6 +345,20 @@ ruleTester.run('prefer-stateless-function', rule, {
       errors: [{
         message: 'Component should be written as a pure function'
       }]
+    }, {
+      code: `
+        class Foo extends React.PureComponent {
+          render() {
+            return <div>this.bar</div>;
```

Comment:
I'm a bit confused here; how can the linter statically know that there's no `bar = foo` defined in the class or the constructor?

Reply:
I see the problem. I was confused by the docs:
>When true the rule will ignore Components extending from React.PureComponent that use this.props or this.context.

The example that you mentioned leads to the fact that this rule will ignore all `this`'s properties, not only `props` and `context`. Is it accurate?

Reply:
If you have a component that doesn't reference props or context, then there's other better mechanisms for anything that hangs off of `this` - so the rule *should* warn on it, and you should refactor to use an SFC.